simphony/cuds/model.py

```diff
@@ -0,0 +1,107 @@
+"""CUDS computational model implementation.
+
+This module contains the classes used to represent a computational model, based on SimPhoNy metadata.
+"""
+from ..core import DataContainer
+from .store import MemoryStateDataStore
+
+
+# IDEA: add consistency check handlers/instances to the CUDS class (dynamic).
+class CUDS(object):
+    """CUDS computational model."""
+    def __init__(self):
+
+        # Add datasets to memory datastore by default
+        self._dataset_store = MemoryStateDataStore()
+
+        # Memory storage to keep CUDS components
+        self._store = {}
+
+        # Data container
+        self._data = DataContainer()
+
+    @staticmethod
+    def _is_cuds_component(obj):
+        """Check whether the object is a CUDS component."""
+        # When I see a bird that walks like a duck and swims like a duck
+        # and quacks like a duck, I call that bird a duck
+        # TODO: replace with metadata type checks
+        return all([hasattr(obj, 'name'),
+                    hasattr(obj, 'data')])
+
+    @property
+    def data(self):
+        """Data container for CUDS items."""
+        return self._data
+
+    @data.setter
+    def data_setter(self):
+        raise Exception('Replacing data container is not supported.')
```

Comment:
Shouldn't we have the same the behaviour as we do with Mesh, Lattice and Particles ítems? Returning a copy of the data, let the user work with the copy, and when he is happy with it set the new data as a copy (screenshot model)